lockfiles/st2-constraints.txt

```diff
@@ -0,0 +1,68 @@
+# Add/remove version constraints for transitive dependencies in this file
+# (transitive dependencies are dependencies of our direct dependencies).
+# Then run `./pants generate-lockfiles --resolve=st2` to regenerate the lockfile.
```

Comment:
Is this still relevant?

Reply:
What do you mean?

- Is this comment relevant? - yes, it describes the "convention" that I'm trying to create to separate transitive constraints from direct requirements.
- Is the file relevant? - yes, but I haven't wired it up in `pants.toml` yet, because that requires configuring resolves and interpreter constraints, both of which I plan to introduce in subsequent PRs.

Reply:
I mean running generate-lockfiles, since that part isn't being used yet. Guess it depends how much adjusting of the prework you want to do - if this change should stand alone vs it truly only is prework for a PR that will be merged soon

Reply:
I dropped the `generate-lockfiles` comment.

Even though this file isn't used yet (not registered in pants.toml) it's very interrelated with the other requirements bits, so I opted to put it in this PR. Wiring up the resolves will come in a subsequent PR.